plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/auth/DefaultAuthCredentialsService.kt

```diff
@@ -66,6 +57,41 @@ class DefaultAuthCredentialsService(
             } ?: completableFuture.completeExceptionally(IllegalStateException("LSP Server not running"))
         }
 
+    override fun onChange(providerId: String, newScopes: List<String>?) {
+        updateTokenFromActiveConnection()
+    }
+
+    override fun activeConnectionChanged(newConnection: ToolkitConnection?) {
+        val qConnection = ToolkitConnectionManager.getInstance(project)
+            .activeConnectionForFeature(QConnection.getInstance())
+            ?: return
+        if (newConnection?.id != qConnection.id) return
+
```

Comment:
non-blocking, but are we sure these lines are not duplicated? I think they already exist somewhere in CW